File: tests/loop/extend_range/loop_extend_range_array_size_doc.cpp
#include <optitrust.h>
#include <stdlib.h>

int main() {
  const int N1 = 10;
  const int *p = MALLOC1(int, N1);
  for (int i = 10; i < 12; i++) {
    i++;
  }
  return 0;
}
// Testing dim parameter
int main2() {
  const int N1 = 10;
  const int N2 = 10;
  const int *p = MALLOC2(int, N1, N2);
  for (int i = 10; i < 12; i++) {
    i++;
  }
}

File: tests/loop/extend_range/loop_extend_range_array_size_doc.ml
open Optitrust
open Prelude

let _ = Run.script_cpp (fun _ ->
  (* !!Matrix.insert_alloc_dim (expr "N2") [ cVarInit "p" ];*)
  !! Loop.extend_range_array_size [ cFunBody "main"; cVarInit "p" ] [ cFunBody "main"; cFor "i" ];
  !! Loop.extend_range_array_size ~dim:1 [ cFunBody "main2"; cVarInit "p" ] [ cFunBody "main2"; cFor "i" ];
)
